### ISSUE TITLE
(given candle is not vaild) . how i can fix it  haki nashi candle

### ISSUE THREAD

**Issue by muroyal** [binary-com/binary-bot#2371]
Getting error while running the bot on forex market .. but seam bot is running fine on synthatic 

Kindly have a look from your end help me..

[Chanikya_haki_v1.0 (2).zip](https://github.com/binary-com/binary-bot/files/4116534/Chanikya_haki_v1.0.2.zip)

**Comment by ZDEGLET** [comment]
У НИХ НА СЕРВЕРАХ КАКАЯ ТО ЕРУНДА. У МЕНЯ НЕКОТОРЫЕ РОБОТЫ ТОЖЕ НЕ РАБОТАЮТ.

**Comment by ZDEGLET** [comment]
ПОДОЖДЕМ ПОСМОТРИМ.

**Comment by hotmatrixx** [comment]
Well, I opened it.

it's a combination of other people's scripts. I'm not invested enough in it to try and fix it for you. What you are facing is a bug - so you need to start pulling it apart piece by piece until it stops posting an error - it's called debugging.

I probably could get tot the bottom of it eventually, but this is a really complex looking issue and tbh, I can only hope that understand the bot fully yourself, based on some of the other threads you've asked requests for.... have you tested that it all works with each separate indicator?

**Comment by muroyal** [comment]
@hotmatrixx the strategy and bot is mine but the bot prepared by @ex machina

**Comment by hotmatrixx** [comment]
> 
> 
> @hotmatrixx the strategy and bot is mine but the bot prepared by @ex machina
> 
> This bot taking more taking more system resources hence I did not spend more time on it...

Yeah, I bet it's using a lot of resources. You've got it doing a complex operation (Henken Ashi candles) for 1000 trades, every tick.

I didn't understand why you had that. Maybe reduce that 999 down to something more reasonable? I stopped poking around at about that point.

Also that henken was coded alright, I built one too but mine was... very different to yours. you have a few doubleup without using loops and lists 'perfectly'. but it works, at first glance.
